### DISCUSSION TITLE
Disabled Text input con alpha 50%

### DISCUSSION THREAD

**Discussion by AnaMontes11** [Telefonica/mistica-design#530] [Proposals]
Al eliminar las constantes `textDisabled` y `textLinkDisabled` para los estados deshabilitados y usar alfa al 50% se detecta que con esta solución los DISABLED TEXT INPUT WITH CONTENT no acaban de funcionar bien porque el input text sigue pareciendo editable (textPrimary --> grey6).

En caso de considerarlos como «readonly» y no como «disabled», tiene sentido que lleve el alfa si es de solo lectura aunque no sea editable? esto daría problemas de accesibilidad. Bajo mi punto de vista, al no entenderse como «disabled» el input text no debería llevar alfa.

Veo que aunque parecidas, son casos de uso diferentes.

![ae159bd9-5ec7-4c05-b74a-f564ee696d47](https://user-images.githubusercontent.com/88187691/175934645-284a5273-77aa-4cbb-8219-3f1dc578397b.jpg)

Original issue: [[ANDROID-10105 Disable states with alpha 50%]](url)

**Comment by yceballost** [comment]
Creo que esto es carne para valorar en el proyecto de Accessibility Analysis https://github.com/Telefonica/mistica-design/discussions/498

**Comment by aweell** [comment]
> En caso de considerarlos como «readonly» y no como «disabled», tiene sentido que lleve el alfa si es de solo lectura aunque no sea editable? esto daría problemas de accesibilidad.

Totalmente, habría que ver luego en qué contexto se utiliza, si es necesario que el texto sea legible pero que la interacción esté deshabilitada no debería usarse el estado disabled, sino que habría que plantear un nuevo estado readonly para este tipo de componentes (Textfield, textarea etc...)

**Comment by yceballost** [comment]
Añado por aquí que el estado `readOnly` en web ya existe
https://ibit.ly/OSeE

**Reply by aweell** [reply]
Dejo por aquí una propuesta para diferenciar ese estado del disabled:

https://codepen.io/AlexBueno/pen/ZExyWBN

**Reply by AnaMontes11** [reply]
Me parece buena solución diferenciar los read-only con background y los disabled además apoyarlos por el not-allowed cursor.
